### PR TITLE
Enhancement (Navigation Header): navigation module (DEV-482)

### PR DESCRIPTION
**Pull request by domsteinbach**
resolves DEV-482

### CODE REVIEW

File: src/NavigationHeader/HeaderModule.elm [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+headerElementsCtStyle =
+    [ Tw.flex
+    , Tw.justify_between
+    , Tw.h_16
+    ]
+        |> css
+
+
+
+-- Container for al left elements
```
Comment (mdelez) [inline src/NavigationHeader/HeaderModule.elm:147]: al -> all

File: src/NavigationHeader/HeaderModule.elm [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+
+
+flexRightElementsCt : List (Attribute msg)
+flexRightElementsCt =
+    [ id "flex-right-elements-ct", attribute "data-tw-class" "flex items-center space-x-4", css [ Tw.flex, Tw.items_center, Tw.space_x_4 ] ]
+
+
+
+-- the mobile menue container. Hidden if the screen size reaches medium size
```
Comment (mdelez) [inline src/NavigationHeader/HeaderModule.elm:206]: menue -> menu

File: src/NavigationHeader/HeaderModule.elm [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+            ]
+        ]
+    ]
+
+
+
+-- colors
+
+
+primary : Css.Color
```
Comment (mdelez) [inline src/NavigationHeader/HeaderModule.elm:226]: are these colors used? You also have the same in `CssColors`

File: src/NavigationHeader/Navitem.elm [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+nItem : List (Attribute msg) -> String -> String -> Cmd msg -> Bool -> Html msg
+nItem attrs text hRef cmd isActive =
+    navItem { attrs = attrs, text = text, hRef = hRef, cmd = cmd, isActive = isActive }
+
+
+type alias NavItem msg =
+    { attrs : List (Attribute msg) -- onClick, disable and all other events as well as custom attributes
+    , text : String
+    , hRef : String -- set to "#" if you like to run sth via onClick
```
Comment (mdelez) [inline src/NavigationHeader/Navitem.elm:18]: hRef -> href

File: src/NavigationHeader/SearchBar.elm [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    , Tw.w_full
+    , Tw.flex
+    , Tw.items_center
+    , Bp.lg [ Tw.max_w_xs ]
+    ]
+        |> css
+
+
+
+-- class="block w-full pl-10 pr-3 py-2 border border-transparent rounded-md leading-5 bg-gray-700 text-gray-300 placeholder-gray-400 focus:outline-none focus:bg-white
```
Comment (mdelez) [inline src/NavigationHeader/SearchBar.elm:46]: can the commented out code be removed?

File: src/NavigationHeader/HeaderModule.elm [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+        , signInButton [] "sign in"
+        ]
+
+
+
+-- attributes & classes ...
+-- the navBar background container
+
+
+navHeaderBgCt : List (Attribute msg)
```
Comment (mdelez) [inline src/NavigationHeader/HeaderModule.elm:73]: It's a small thing, but using `ct` as the shorthand for `container` isn't really clear. Perhaps `cntr` would be better.

**Review by mdelez — CHANGES_REQUESTED** [review]
It's a good first implementation but we should all have a meeting about how we are going to implement CSS

File: src/NavigationHeader/HeaderModule.elm (hunk 6 above)
Comment (domsteinbach) [reply src/NavigationHeader/HeaderModule.elm:73]: good with that. It is an eary draft though, hence all css and how we implement it is not yet clear and how the buttons needed do not yet support anything.

**Review by mdelez — APPROVED** [review]
Thanks!

**Pull request merged**